bin/strata-client/src/rpc_server.rs

```diff
@@ -138,12 +141,11 @@ impl<D: Database + Send + Sync + 'static> StrataApiServer for StrataRpcImpl<D> {
     }
 
     async fn get_clientstate_at_idx(&self, idx: u64) -> RpcResult<Option<ClientState>> {
-        let db = self.database.clone();
-        let client_state = wait_blocking("client_state_at_idx", move || {
-            db.client_state_db().get_state_at(idx).map_err(Error::Db)
-        })
-        .await?;
-        Ok(client_state)
+        let database = self.database.clone();
+        let client_state_db = database.client_state_db().as_ref();
+        let cs = reconstruct_state(client_state_db, idx)
```

Comment:
This is blocking, still needs to wrap it in `wait_blocking`.  Eventually this would be replaced with the client state manager which does the reconstruction internally.